### PR TITLE
Add `DiMuonVertexMonitor` module and use it for `ALCARECOTkAlDiMuonAndVertexDQM`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

PR  https://github.com/cms-sw/cmssw/pull/33770 added a new ALCARECO producer (`ALCARECOTkAlDiMuonAndVertex`).
This PR adds the DQM monitoring for that stream and adds a new DQM Analyzer (`DiMuonVertexMonitor`)  to monitor the properties of the selected di-muon vertices.
I profit of this PR to make a clean-up of the `TkAlCaRecoMonitor` module as well in commit 95e18ef2d8be85a5452c2ec612ce844c9e9bd5ab.

#### PR validation:

Run successfully a modified version of 136.887 to include the `TkAlDiMuonAndVertex` ALCARECO producer.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport, but should probably be backported to 12.4.X

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: DQMOffline/Alignment/python/ALCARECOTkAlDQM_cff.py [hunk 1]
```diff
@@ -57,6 +58,40 @@
 #ALCARECOTkAlZMuMuDQM = cms.Sequence( ALCARECOTkAlZMuMuTrackingDQM + ALCARECOTkAlZMuMuTkAlDQM + ALCARECOTkAlZMuMuHLTDQM )
```
Comment (emanueleusai) [inline DQMOffline/Alignment/python/ALCARECOTkAlDQM_cff.py:58]: Are all the commented code lines in this file necessary? I understand they were not added in this PR but maybe we can clean up a bit...

Comment (mmusich) [reply DQMOffline/Alignment/python/ALCARECOTkAlDQM_cff.py:58]: Given the time necessary for the re-testing, review, etc., and considering this needs to be backported I would prefer to clean up in a separate PR if there are no comments on the part added here.

Comment (emanueleusai) [reply DQMOffline/Alignment/python/ALCARECOTkAlDQM_cff.py:58]: Fair enough

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**